src/Magento/FunctionalTestingFramework/Util/TestGenerator.php

```diff
@@ -774,6 +813,41 @@ public function generateStepsPhp($actionObjects, $generationScope = TestGenerato
             }
 
             switch ($actionObject->getType()) {
+                case "helper":
+                    if (!in_array($customActionAttributes['class'], $this->customHelpers)) {
+                        $this->customHelpers['$' . $stepKey] = $customActionAttributes['class'];
+                    }
+
+                    $arguments = [];
+                    $classReader = new \Magento\FunctionalTestingFramework\Helper\Code\ClassReader();
+                    $parameters = $classReader->getParameters(
+                        $customActionAttributes['class'],
+                        $customActionAttributes['method']
+                    );
+                    $typesToQuote = ['string', 'float'];
+                    foreach ($parameters as $parameter) {
+                        if (isset($customActionAttributes[$parameter['variableName']])) {
+                            $value = $customActionAttributes[$parameter['variableName']];
+                            $arguments[] = $this->addUniquenessFunctionCall(
+                                $value,
+                                in_array($parameter['type'], $typesToQuote) || $parameter['type'] === null
+                            );
+                        } elseif (!$parameter['isOptional']) {
+                            throw new TestFrameworkException(
+                                'Argument \'' . $parameter['variableName'] . '\' for method '
+                                . $customActionAttributes['class'] . '::' . $customActionAttributes['method']
+                                . ' is not found.'
+                            );
+                        }
+                    }
```

Comment:
When helper method use some optional arguments, arguments don't resolve correctly. For example, I try to set "bla", no matter if I set "superBla" to "null" or not, the generated code is not correct.
{code}
            <argument name="superBla">null</argument>
            <argument name="bla">1.2</argument>
{code}

Reply:
Good catch! Fixed

Reply:
After the fix, what's syntax to use if I want to pass in `null`? what's the syntax to use if i want to pass in a float `1.2`?

Reply:
```
<argument name="superBla">1.2</argument>
<argument name="bla" />
```
it will generate: 
```
$this->helperContainer->get('\MFTF\DevDocs\Helper\CustomHelper')->goTo("mftf/docs/introduction.html", ".page-intro", ['.page-intro', 'Bla'], "1.2", "", array (  'key' => 'value',  0 => 'test',)); // stepKey: customHelperWithArrayParametrized

```

I intentionally do not give an opportunity to pass null from Test because I don't any possible scenario someone would need it. 
But I see scenario when I would want to pass empty string. For writing tests I think it is useful. But still if there is a case when you need `null`, specify it as optional value in class and do not pass this argument.

Reply:
Verified it worked now